applications/dashboard/views/modules/me.php

```diff
@@ -59,51 +59,37 @@
-    echo sprite('SpFlyoutHandle', 'Arrow');
-    echo '<div class="Flyout MenuItems">';
-    echo '<ul>';
-    // echo wrap(Wrap(t('My Account'), 'strong'), 'li');
-    // echo wrap('<hr />', 'li');
-    if (hasEditProfile(Gdn::session()->UserID)) {
-        echo wrap(Anchor(sprite('SpEditProfile').' '.t('Edit Profile'), 'profile/edit', 'EditProfileLink'), 'li', array('class' => 'EditProfileWrap link-editprofile'));
-    } else {
-        echo wrap(Anchor(sprite('SpEditProfile').' '.t('Preferences'), 'profile/preferences', 'EditProfileLink'), 'li', array('class' => 'EditProfileWrap link-preferences'));
-    }
-
-    if ($Session->checkPermission(array('Garden.Settings.View', 'Garden.Settings.Manage', 'Garden.Moderation.Manage', 'Garden.Users.Approve', 'Moderation.Spam.Manage', 'Moderation.ModerationQueue.Manage'), false)) {
-        echo wrap('<hr />', 'li');
-        $CApplicant = $ApplicantCount > 0 ? ' '.Wrap($ApplicantCount, 'span class="Alert"') : '';
-        $CSpam = ''; //$SpamCount > 0 ? ' '.Wrap($SpamCount, 'span class="Alert"') : '';
-        $CModeration = $ModerationCount > 0 ? ' '.Wrap($ModerationCount, 'span class="Alert"') : '';
-
-        if ($Session->checkPermission('Garden.Users.Approve')) {
-            echo wrap(Anchor(sprite('SpApplicants').' '.t('Applicants').$CApplicant, '/dashboard/user/applicants'), 'li', array('class' => 'link-applicants'));
-        }
-
-        if ($Session->checkPermission(array('Garden.Settings.Manage', 'Garden.Moderation.Manage', 'Moderation.ModerationQueue.Manage'), false)) {
-            echo wrap(Anchor(sprite('SpSpam').' '.t('Spam Queue').$CSpam, '/dashboard/log/spam'), 'li', array('class' => 'link-spam'));
-        }
-
-        if ($Session->checkPermission(array('Garden.Settings.Manage', 'Garden.Moderation.Manage', 'Moderation.ModerationQueue.Manage'), false)) {
-            echo wrap(Anchor(sprite('SpMod').' '.t('Moderation Queue').$CModeration, '/dashboard/log/moderation'), 'li', array('class' => 'link-moderation'));
-        }
-
-        if ($Session->checkPermission(array('Garden.Settings.View', 'Garden.Settings.Manage'), false)) {
-            echo wrap(Anchor(sprite('SpDashboard').' '.t('Dashboard'), '/dashboard/settings'), 'li', array('class' => 'link-dashboard'));
-        }
-    }
-
+    $dropdown = new DropdownModule();
+    $dropdown->setData('DashboardCount', $DashboardCount);
+    $dropdown->setTrigger(wrap(t('Account Options'), 'em'), 'anchor', 'MeButton FlyoutButton', '<span class="Sprite Sprite16 SpOptions"></span>', '/profile/edit');
+    $editModifiers['listItemCssClasses'] = ['EditProfileWrap', 'link-editprofile'];
+    $preferencesModifiers['listItemCssClasses'] = ['EditProfileWrap', 'link-preferences'];
+
+    $dropdown->addLinkIf(hasEditProfile(Gdn::session()->UserID), t('Edit Profile'), '/profile/edit', 'profile.edit', '', [], $editModifiers);
```

Comment:
This fails because `hasEditProfile()` return `(int)1` instead of `(bool)true`.
`hasEditProfile` needs to be fixes before I can merge this!